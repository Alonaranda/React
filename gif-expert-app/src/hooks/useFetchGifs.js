import {useState,useEffect} from 'react';
import {getGifs} from '../helpers/getGifs';

//Son como functional component

export const useFetchGifs= (category) => {
    const [state, setState] = useState({
        data: [],
        loading: true
    });

    useEffect(()=>{
        getGifs(category).then(imgs =>{setState({
            data:imgs,
            loading: false
        })})
    },[category])

    return state;
}