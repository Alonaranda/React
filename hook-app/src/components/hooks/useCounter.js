import {useState} from 'react'


export const useCounter = (initialState = 0) => {

    //usamos el useState
    const [counter, setCounter] = useState(initialState);

    //Incrementar
    const increment = () => {
        setCounter(counter + 1);
    }

    //Decrementar
    const decrement = () =>{
        setCounter(counter - 1);
    } 

    //Reset
    const reset = () => {
        setCounter(initialState);
    }

    return {
        counter,
        increment,
        decrement, 
        reset
    };
}
