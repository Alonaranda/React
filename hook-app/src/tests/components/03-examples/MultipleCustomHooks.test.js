import { MultipleCustomHooks } from "../../../components/03-examples/MultipleCustomHooks";
import React from 'react';
import { shallow} from 'enzyme';
import { useFetch } from "../../../hooks/useFetch";
//import { useFetch } from "../../../hooks/useFetch";
jest.mock('../../../hooks/useFetch');


describe('Probando MultipleCustomHook', () => {


    test('Debe de mostrrase correctamente ', () => {
        useFetch.mockReturnValue({
            data: null,
            loding: true,
            error: null
        })
        const wrapper = shallow(<MultipleCustomHooks/>);
        expect(wrapper).toMatchSnapshot();
    });

    test('Debe de mostrar la informacion ', () => {
        useFetch.mockReturnValue({
            data: [{
                data: [{
                    author: 'Chris',
                    quote: 'Hola mundo'
                }],
                loding: false,
                error: null
            }],
            loding: true,
            error: null
        });
        const wrapper = shallow(<MultipleCustomHooks/>);
        expect(wrapper.find('.alert').exists()).toBe(false);
        //expect(wrapper.find('.mb-2').text().trim()).toBe('Hola mundo');
        //expect(wrapper.find('footer').text().trim()).toBe('Chris');
        //console.log(wrapper.html());
    });
});