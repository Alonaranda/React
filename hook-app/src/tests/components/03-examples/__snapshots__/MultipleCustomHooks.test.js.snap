// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Probando MultipleCustomHook Debe de mostrrase correctamente  1`] = `
<div>
  <h1>
    BreakingBad Quotes
  </h1>
  <hr />
  <blockquote
    className="blockquote text-right"
  >
    <p
      className="mb-2"
    />
    <footer
      className="blockquote-footer"
    />
  </blockquote>
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Siguiente
  </button>
</div>
`;
